_posts/2015-02-06-Utilising-API-Blueprint-in-API-Clients.md

```diff
@@ -0,0 +1,119 @@
+
+This allows us to decouple an API client from the server-side implementation, we no longer have to know about the URIs to each resource and the HTTP methods in our code. We can negotiate these at run-time, similar to how with HATEOS we follow links, we can follow links from an API Blueprint.
+
+## Example
+
+Let’s walk though how to load an API Blueprint and perform transitions in Swift. First of all, we’re going to load our blueprint file that’s embedded in our application. You’re not limited to embedding it in your application. You could even retrieve it dynamically at run-time.
+
+```swift
+let blueprint = Blueprint(named: "blueprint.json", bundle: nil)
+```
+
+Once you’ve loaded your blueprint, you can retrieve the different actions as [transitions](https://github.com/the-hypermedia-project/charter/blob/master/reference/hypermedia-elements.md#transitions) from your blueprint.
+
+This example is going to make use of one of our example blueprints, “Polls API”. I’ve pulled out an excerpt of the example blueprint and it looks something like the following. You can find the full version over in our [examples](https://github.com/apiaryio/api-blueprint/blob/kylef/polls-example/examples/Polls%20API.md).
+
+```markdown
+# Polls
+Polls is a simple web service that allows consumers to view polls and vote in them.
+
+## Group Question
+Resource related to questions in the API.
+
+## Question [/questions/{question_id}]
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+
+### View a question detail [GET]
+
+## Choice [/questions/{question_id}/choices/{choice_id}]
+
++ Parameters
+    + question_id (number) ... ID of the Question in form of an integer
+    + choice_id (number) ... ID of the Choice in form of an integer
+
+### Vote [PUT]
+
+This action allows you to vote on a question's choice.
+
+## Questions collection [/questions{?page}]
++ Parameters
+    + page (optional, number) ... The page of questions to return
+
+### List all questions [GET]
+### Create a new question [POST]
+```
+
+We can pull out any of the actions from our blueprint by the resource and action name in Swift:
+
+```swift
+if let transition = blueprint?.transition("Choice", action: "Vote") {
+}
+```
+
+This transition allows us to introspect it’s values at run-time, allowing us to retrieve protocol information such as the HTTP method and URI along with all the URI parameters.
+
+```swift
+println(transition.uri)
+println(transition.method)
+
+for (name, parameter) in transition.parameters {
+  println(name)
+  println("\(parameter.value) \(parameter.defaultValue)")
+}
+```
+
+We can use the elegant Swift HTTP networking library [Alamofire](https://github.com/Alamofire/Alamofire) along with some convenience methods we’ve built to perform HTTP requests from our transition directly.
+
+```swift
+let baseURL = NSURL(string: "http://polls.apiblueprint.org/")
```

Comment:
Out of curiosity – isn't that root URL hidden somewhere in the blueprint Adapter?